sympy/combinatorics/free_groups.py

```diff
@@ -93,11 +90,13 @@ def _parse_symbols(symbols):
     if not symbols:
         return tuple()
     if isinstance(symbols, string_types):
+        from sympy.core.symbol import symbols as _symbols
         return _symbols(symbols, seq=True)
```

Comment:
This creates sympy symbols.

Reply:
From what I observe, `SymEngine.py`'s `symbols` has the same code as `SymPy`, still this is raising issues.

Reply:
You mentioned issues previously as well. Care to elaborate?

Reply:
Here is the complete error log I obtained:
```
File "/home/shikhar/sympy/sympy/combinatorics/tests/test_coset_table.py", line 141, in test_coset_enumeration
    E1 = FpGroup(F, [t**-1*r*t*r**-2, r**-1*s*r*s**-2, s**-1*t*s*t**-2])
  File "/home/shikhar/sympy/sympy/combinatorics/fp_groups.py", line 75, in __init__
    self._rewriting_system = RewritingSystem(self)
  File "/home/shikhar/sympy/sympy/combinatorics/rewritingsystem.py", line 36, in __init__
    self._init_rules()
  File "/home/shikhar/sympy/sympy/combinatorics/rewritingsystem.py", line 62, in _init_rules
    self._remove_redundancies()
  File "/home/shikhar/sympy/sympy/combinatorics/rewritingsystem.py", line 153, in _remove_redundancies
    v = self.reduce(r, exclude=r)
  File "/home/shikhar/sympy/sympy/combinatorics/rewritingsystem.py", line 282, in reduce
    new = new.eliminate_word(r, rules[r], _all=True, inverse=False)
  File "/home/shikhar/sympy/sympy/combinatorics/free_groups.py", line 659, in eliminate_word
    return word.eliminate_word(gen, by, _all=True, inverse=inverse)
  File "/home/shikhar/sympy/sympy/combinatorics/free_groups.py", line 659, in eliminate_word
    return word.eliminate_word(gen, by, _all=True, inverse=inverse)
  File "/home/shikhar/sympy/sympy/combinatorics/free_groups.py", line 659, in eliminate_word
    return word.eliminate_word(gen, by, _all=True, inverse=inverse)
  [Previous line repeated 965 more times]
  File "/home/shikhar/sympy/sympy/combinatorics/free_groups.py", line 656, in eliminate_word
    word = word.subword(0, i)*by**k*word.subword(i+l, len(word)).eliminate_word(gen, by)
  File "/home/shikhar/sympy/sympy/combinatorics/free_groups.py", line 635, in eliminate_word
    if self.is_independent(gen) or gen == by:
  File "/home/shikhar/sympy/sympy/combinatorics/free_groups.py", line 948, in is_independent
    return not self.is_dependent(word)
  File "/home/shikhar/sympy/sympy/combinatorics/free_groups.py", line 936, in is_dependent
    return self.subword_index(word**-1) != None
  File "/home/shikhar/sympy/sympy/combinatorics/free_groups.py", line 472, in __pow__
    return (self.inverse())**n
  File "/home/shikhar/sympy/sympy/combinatorics/free_groups.py", line 546, in inverse
    r = tuple([(i, -j) for i, j in self.array_form[::-1]])
  File "/home/shikhar/sympy/sympy/combinatorics/free_groups.py", line 378, in array_form
    return tuple(self)
  File "/home/shikhar/sympy/sympy/combinatorics/free_groups.py", line 681, in __len__
    return sum(abs(j) for (i, j) in self)
RecursionError: maximum recursion depth exceeded
_______________________________________________________________________________________________________________________________________________________________________________________
_______________________________________________________ sympy/combinatorics/tests/test_fp_groups.py:test_subgroup_presentations _______________________________________________________
  File "/home/shikhar/sympy/sympy/combinatorics/tests/test_fp_groups.py", line 116, in test_subgroup_presentations
    p1 = reidemeister_presentation(f, H)
  File "/home/shikhar/sympy/sympy/combinatorics/fp_groups.py", line 1245, in reidemeister_presentation
    gens, rels = simplify_presentation(gens, rels)
  File "/home/shikhar/sympy/sympy/combinatorics/fp_groups.py", line 925, in simplify_presentation
    F = free_group(syms)[0]
  File "/home/shikhar/sympy/sympy/combinatorics/free_groups.py", line 37, in free_group
    _free_group = FreeGroup(symbols)
  File "/home/shikhar/sympy/sympy/combinatorics/free_groups.py", line 139, in __new__
    symbols = tuple(_parse_symbols(symbols))
  File "/home/shikhar/sympy/sympy/combinatorics/free_groups.py", line 102, in _parse_symbols
    raise ValueError("The type of `symbols` must be one of the following: "
ValueError: The type of `symbols` must be one of the following: a str, Symbol/Expr or a sequence of one of these types
_______________________________________________________________________________________________________________________________________________________________________________________
_______________________________________________________________ sympy/combinatorics/tests/test_fp_groups.py:test_order ________________________________________________________________
  File "/home/shikhar/sympy/sympy/combinatorics/tests/test_fp_groups.py", line 153, in test_order
    assert f.order() == 8
  File "/home/shikhar/sympy/sympy/combinatorics/fp_groups.py", line 225, in order
    self._order = ind*self.subgroup(gens, C=C).order()
  File "/home/shikhar/sympy/sympy/combinatorics/fp_groups.py", line 131, in subgroup
    g, rels = reidemeister_presentation(self, gens, C=C)
  File "/home/shikhar/sympy/sympy/combinatorics/fp_groups.py", line 1245, in reidemeister_presentation
    gens, rels = simplify_presentation(gens, rels)
  File "/home/shikhar/sympy/sympy/combinatorics/fp_groups.py", line 925, in simplify_presentation
    F = free_group(syms)[0]
  File "/home/shikhar/sympy/sympy/combinatorics/free_groups.py", line 37, in free_group
    _free_group = FreeGroup(symbols)
  File "/home/shikhar/sympy/sympy/combinatorics/free_groups.py", line 139, in __new__
    symbols = tuple(_parse_symbols(symbols))
  File "/home/shikhar/sympy/sympy/combinatorics/free_groups.py", line 102, in _parse_symbols
    raise ValueError("The type of `symbols` must be one of the following: "
ValueError: The type of `symbols` must be one of the following: a str, Symbol/Expr or a sequence of one of these types
_______________________________________________________________________________________________________________________________________________________________________________________
________________________________________________________ sympy/combinatorics/tests/test_fp_groups.py:test_permutation_methods _________________________________________________________
  File "/home/shikhar/sympy/sympy/combinatorics/tests/test_fp_groups.py", line 177, in test_permutation_methods
    T = G._to_perm_group()[1]
  File "/home/shikhar/sympy/sympy/combinatorics/fp_groups.py", line 365, in _to_perm_group
    if self.order() == S.Infinity:
  File "/home/shikhar/sympy/sympy/combinatorics/fp_groups.py", line 225, in order
    self._order = ind*self.subgroup(gens, C=C).order()
  File "/home/shikhar/sympy/sympy/combinatorics/fp_groups.py", line 131, in subgroup
    g, rels = reidemeister_presentation(self, gens, C=C)
  File "/home/shikhar/sympy/sympy/combinatorics/fp_groups.py", line 1245, in reidemeister_presentation
    gens, rels = simplify_presentation(gens, rels)
  File "/home/shikhar/sympy/sympy/combinatorics/fp_groups.py", line 925, in simplify_presentation
    F = free_group(syms)[0]
  File "/home/shikhar/sympy/sympy/combinatorics/free_groups.py", line 37, in free_group
    _free_group = FreeGroup(symbols)
  File "/home/shikhar/sympy/sympy/combinatorics/free_groups.py", line 139, in __new__
    symbols = tuple(_parse_symbols(symbols))
  File "/home/shikhar/sympy/sympy/combinatorics/free_groups.py", line 102, in _parse_symbols
    raise ValueError("The type of `symbols` must be one of the following: "
ValueError: The type of `symbols` must be one of the following: a str, Symbol/Expr or a sequence of one of these types
_______________________________________________________________________________________________________________________________________________________________________________________
_______________________________________________________ sympy/combinatorics/tests/test_free_groups.py:test_FreeGroup__getitem__ _______________________________________________________
  File "/home/shikhar/sympy/sympy/combinatorics/tests/test_free_groups.py", line 60, in test_FreeGroup__getitem__
    assert F[0:] == FreeGroup("x, y, z")
  File "/home/shikhar/sympy/sympy/combinatorics/free_groups.py", line 210, in __getitem__
    return self.clone(symbols=symbols)
  File "/home/shikhar/sympy/sympy/combinatorics/free_groups.py", line 182, in clone
    return self.__class__(symbols or self.symbols)
  File "/home/shikhar/sympy/sympy/combinatorics/free_groups.py", line 139, in __new__
    symbols = tuple(_parse_symbols(symbols))
  File "/home/shikhar/sympy/sympy/combinatorics/free_groups.py", line 102, in _parse_symbols
    raise ValueError("The type of `symbols` must be one of the following: "
ValueError: The type of `symbols` must be one of the following: a str, Symbol/Expr or a sequence of one of these types
_______________________________________________________________________________________________________________________________________________________________________________________
__________________________________________________________ sympy/combinatorics/tests/test_homomorphisms.py:test_homomorphism __________________________________________________________
  File "/home/shikhar/sympy/sympy/combinatorics/tests/test_homomorphisms.py", line 18, in test_homomorphism
    assert T.is_isomorphism()
  File "/home/shikhar/sympy/sympy/combinatorics/homomorphisms.py", line 208, in is_isomorphism
    return self.is_injective() and self.is_surjective()
  File "/home/shikhar/sympy/sympy/combinatorics/homomorphisms.py", line 188, in is_injective
    return self.kernel().order() == 1
  File "/home/shikhar/sympy/sympy/combinatorics/homomorphisms.py", line 108, in kernel
    self._kernel = self._compute_kernel()
  File "/home/shikhar/sympy/sympy/combinatorics/homomorphisms.py", line 114, in _compute_kernel
    G_order = G.order()
  File "/home/shikhar/sympy/sympy/combinatorics/fp_groups.py", line 225, in order
    self._order = ind*self.subgroup(gens, C=C).order()
  File "/home/shikhar/sympy/sympy/combinatorics/fp_groups.py", line 131, in subgroup
    g, rels = reidemeister_presentation(self, gens, C=C)
  File "/home/shikhar/sympy/sympy/combinatorics/fp_groups.py", line 1245, in reidemeister_presentation
    gens, rels = simplify_presentation(gens, rels)
  File "/home/shikhar/sympy/sympy/combinatorics/fp_groups.py", line 925, in simplify_presentation
    F = free_group(syms)[0]
  File "/home/shikhar/sympy/sympy/combinatorics/free_groups.py", line 37, in free_group
    _free_group = FreeGroup(symbols)
  File "/home/shikhar/sympy/sympy/combinatorics/free_groups.py", line 139, in __new__
    symbols = tuple(_parse_symbols(symbols))
  File "/home/shikhar/sympy/sympy/combinatorics/free_groups.py", line 102, in _parse_symbols
    raise ValueError("The type of `symbols` must be one of the following: "
ValueError: The type of `symbols` must be one of the following: a str, Symbol/Expr or a sequence of one of these types
_______________________________________________________________________________________________________________________________________________________________________________________
___________________________________________________________ sympy/combinatorics/tests/test_perm_groups.py:test_presentation ___________________________________________________________
  File "/home/shikhar/sympy/sympy/combinatorics/tests/test_perm_groups.py", line 837, in test_presentation
    assert _test(P)
  File "/home/shikhar/sympy/sympy/combinatorics/tests/test_perm_groups.py", line 833, in _test
    G = P.presentation()
  File "/home/shikhar/sympy/sympy/combinatorics/perm_groups.py", line 3799, in presentation
    G_p = _factor_group_by_rels(G_p, [new_rel])
  File "/home/shikhar/sympy/sympy/combinatorics/perm_groups.py", line 3709, in _factor_group_by_rels
    g for g in G.relators + rels)))
  File "/home/shikhar/sympy/sympy/combinatorics/fp_groups.py", line 75, in __init__
    self._rewriting_system = RewritingSystem(self)
  File "/home/shikhar/sympy/sympy/combinatorics/rewritingsystem.py", line 36, in __init__
    self._init_rules()
  File "/home/shikhar/sympy/sympy/combinatorics/rewritingsystem.py", line 62, in _init_rules
    self._remove_redundancies()
  File "/home/shikhar/sympy/sympy/combinatorics/rewritingsystem.py", line 154, in _remove_redundancies
    w = self.reduce(rules[r])
  File "/home/shikhar/sympy/sympy/combinatorics/rewritingsystem.py", line 282, in reduce
    new = new.eliminate_word(r, rules[r], _all=True, inverse=False)
  File "/home/shikhar/sympy/sympy/combinatorics/free_groups.py", line 659, in eliminate_word
    return word.eliminate_word(gen, by, _all=True, inverse=inverse)
  File "/home/shikhar/sympy/sympy/combinatorics/free_groups.py", line 659, in eliminate_word
    return word.eliminate_word(gen, by, _all=True, inverse=inverse)
  File "/home/shikhar/sympy/sympy/combinatorics/free_groups.py", line 659, in eliminate_word
    return word.eliminate_word(gen, by, _all=True, inverse=inverse)
  [Previous line repeated 962 more times]
  File "/home/shikhar/sympy/sympy/combinatorics/free_groups.py", line 656, in eliminate_word
    word = word.subword(0, i)*by**k*word.subword(i+l, len(word)).eliminate_word(gen, by)
  File "/home/shikhar/sympy/sympy/combinatorics/free_groups.py", line 635, in eliminate_word
    if self.is_independent(gen) or gen == by:
  File "/home/shikhar/sympy/sympy/combinatorics/free_groups.py", line 948, in is_independent
    return not self.is_dependent(word)
  File "/home/shikhar/sympy/sympy/combinatorics/free_groups.py", line 936, in is_dependent
    return self.subword_index(word**-1) != None
  File "/home/shikhar/sympy/sympy/combinatorics/free_groups.py", line 472, in __pow__
    return (self.inverse())**n
  File "/home/shikhar/sympy/sympy/combinatorics/free_groups.py", line 546, in inverse
    r = tuple([(i, -j) for i, j in self.array_form[::-1]])
  File "/home/shikhar/sympy/sympy/combinatorics/free_groups.py", line 378, in array_form
    return tuple(self)
  File "/home/shikhar/sympy/sympy/combinatorics/free_groups.py", line 681, in __len__
    return sum(abs(j) for (i, j) in self)
RecursionError: maximum recursion depth exceeded

_______________________________________________________________________________________________________________________________________________________________________________________
____________________________________________________________ sympy/combinatorics/tests/test_fp_groups.py:test_fp_subgroup _____________________________________________________________
  File "/home/shikhar/sympy/sympy/combinatorics/tests/test_fp_groups.py", line 169, in test_fp_subgroup
    assert x**-1*y**4*x in S
AssertionError
_______________________________________________________________________________________________________________________________________________________________________________________
___________________________________________________ sympy/combinatorics/tests/test_free_groups.py:test_FreeGroupElm_eliminate_word ____________________________________________________
  File "/home/shikhar/sympy/sympy/combinatorics/tests/test_free_groups.py", line 106, in test_FreeGroupElm_eliminate_word
    assert w3.eliminate_word(y, x**-1) == x**-3
AssertionError
_______________________________________________________________________________________________________________________________________________________________________________________
_____________________________________________________________ sympy/combinatorics/tests/test_rewriting.py:test_rewriting ______________________________________________________________
  File "/home/shikhar/sympy/sympy/combinatorics/tests/test_rewriting.py", line 9, in test_rewriting
    assert R.is_confluent
AssertionError
```

Reply:
```
ValueError: The type of `symbols` must be one of the following: a str, Symbol/Expr or a sequence of one of these types
```
Above is because of the `Expr` issue

Reply:
Though importing `Expr` should solve the exceptions of the value error type.

Reply:
Did you try with importing `Expr`? The other errors might be indirect issues. Please fix all the issues mentioned and report back if there are any issues.